WpfTest/ViewModel/LoginViewModel.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Windows.Input;
+using WpfTest.Commands;
+using WpfTest.View;
+
+namespace WpfTest.ViewModel
+{
+    public class LoginViewModel : BaseViewModel
+    {
+        private readonly LoginValidator _loginValidator;
+        private string _login;
+        private string _password;
+        private string _loginError;
+        private bool _isLoginSuccessful;
+
+        public LoginViewModel()
+        {
+            _loginValidator = new LoginValidator();
+            LoginCommand = new RelayCommand(OnLogin);
+        }
+
+        public string Login
+        {
+            get => _login;
+            set
+            {
+                _login = value;
+                OnPropertyChanged("Login");
+            }
+        }
+
+        public string Password
+        {
+            get => _password;
+            set
+            {
+                _password = value;
+                OnPropertyChanged("Password");
+            }
+        }
+
+        public string LoginError
+        {
+            get => _loginError;
+            set
+            {
+                _loginError = value;
+                OnPropertyChanged("LoginError");
+            }
+        }
+
+        public bool IsLoginSuccessful
+        {
+            get => _isLoginSuccessful;
+            set
+            {
+                _isLoginSuccessful = value;
+                OnPropertyChanged("IsLoginSuccessful");
+            }
+        }
+
+        public ICommand LoginCommand { get; }
+
+        private void OnLogin(object parameter)
+        {
+            var results = _loginValidator.Validate(this);
```

Comment:
you should validate not ViewModel but specific Model, so in you case is Login Model like this: The main idea to use LoginModel class as property it in xaml:

```xaml
<TextBox Text="{Binding LoginModel.Login,UpdateSourceTrigger=PropertyChanged, ValidatesOnDataErrors=True}"

<components:BindablePasswordBox  Password="{Binding LoginModel.Password}" BorderThickness="0" Background="Transparent" />
```



```C#

public LoginModel 
{
            get => _loginModel;
            set
            {
                   _loginModel = value
                   OnPropertyChanged("LoginModel");
            }
}

public void OnLogin(object obj)
{
     var results = _loginValidator.Validate(LoginModel);
     // ....
}
```